### PR TITLE
Update sources to make use of vinyl-fs

### LINKED ISSUE #18: Output files to the file system
Currently, the files are simply stored in a `Map` and printed. Next step is to really write them to the file system.

### PR DESCRIPTION
**Pull request by samvv**
- Fixes #18 by performing a `vfs.pipe(dest)`
 - Cleaned up some unused files and imports
 - Slightly new API using a `documenter.analyse()`-step and `documenter.run()` returning a `through2`-stream working on vinyl files
 - Also includes #33

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 language: node_js
+before_script:
+- npm run prepare
```
Comment (BurtHarris) [inline .travis.yml]: I thought that was already happening.   Perhaps it should be `npm install` instead?

File: package.json [hunk 2]
```diff
@@ -32,14 +32,18 @@
   },
   "homepage": "https://github.com/TypeForce/tsdoc#readme",
   "dependencies": {
+    "@types/vinyl": "^2.0.0",
```
Comment (BurtHarris) [inline package.json]: @types dependencies should go into devDependencies

File: .travis.yml (hunk 1 above)
Comment (samvv) [inline .travis.yml]: Yes, I was thinking the same thing, but I checked the logs and it didn't happen. Maybe a bug with `npm`?

File: package.json (hunk 2 above)
Comment (samvv) [inline package.json]: They are needed when the application is [used as a library](https://github.com/Microsoft/types-publisher/issues/81#issuecomment-234051345), but I guess it doesn't matter much.

**Pull request merged**